### ISSUE TITLE
Self hosted updates since March have resulted in games that previously worked reasonably well to stop working

### ISSUE THREAD

**Issue by freeworldgroup** [ruffle-rs/ruffle#3525]
**This is bug appears with the self-hosted version of games on freeworldgroup.com - played on a Windows 10 Alienware laptop, Chrome browser.** 

All March updates result in games that worked with previous versions of RUFFLE to not work at all - getting stuck on menus or loading screens. Not all games are affected, but some that were impacted include:

http://www.freeworldgroup.com/games6/gameindex/electricman-2-hs.htm
http://www.freeworldgroup.com/games6/gameindex/missile-game-3d-hs.htm
http://www.freeworldgroup.com/games6/gameindex/big-bucks.htm

After reverting back to the 28 Feb update, the games seem to function ok again. Hoping whatever change happened can be addressed. 

have attached the swf files to the above games. 
thanks

[gamefiles.zip](https://github.com/ruffle-rs/ruffle/files/6081573/gamefiles.zip)

**Comment by Poopooracoocoo** [comment]
Could be the same as #3501?

**Comment by torokati44** [comment]
For at least Electricman2HS, the regression was caused by dad21d43, as it was already suggested by someone on Discord.

**Comment by torokati44** [comment]
At least for Electricman2HS (so, for AVM1), this substitution (both in `action_jump()` and `action_if`) in `core/src/avm1/activation.rs` fixes the issue:
```diff
- reader.seek(data.as_ref(), jump_offset);
+ reader.seek(data.movie.data(), jump_offset);
```

It looks to me that the entire movie data has to be passed in, instead of only the current slice.
Could you please check if this could be easily ported to the avm2 part of the dad21d4 as well, @madsmtm ?

**Comment by torokati44** [comment]
Fixed by #3566.

**Comment by madsmtm** [comment]
Thanks for finding and fixing this issue, and sorry for not fixing it myself as I said I would on Discord 😬
